### PR TITLE
Fix unhandled ValueError raised during handshake

### PR DESCRIPTION
**Pull request by carver**
### What was wrong?

An invalid public key was causing an uncaught `ValueError` during handshake

### How was it fixed?

Catch the `ValueError` raised by the encryption library on a bad public key, and re-raise a `DecryptionError`.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by pipermerriam** [comment]
Meta comment, not requesting anything in this PR.  I would like for us to find a way to test these things.  We have probably 20+ of these patches/fixes that have gone in, catching various edge case exceptions that show up when peers send invalid information.  Our strategy for fixing them is fine but I suspect we are very quickly approaching a point where some of these will do one of the following:

- become a dead code path due to refactor resulting in the exception scenario not being possible anymore.
- masque an error in our implementation

The later of these actually doesn't worry me much because I suspect most of those cases will manifest as problems elsewhere with other clients disconnecting us.

However, if they become dead code paths, we'll just be living with that dead code forever which is lame.  Solution seems to be trying to figure out an avenue to test these, but I suspect those will end up being very hard to read tests that are hard to maintain and potentially fragile.

File: p2p/ecies.py [hunk 1]
```diff
@@ -100,7 +100,12 @@ def decrypt(data: bytes, privkey: datatypes.PrivateKey, shared_mac_data: bytes =
 
     #  1) generate shared-secret = kdf( ecdhAgree(myPrivKey, msg[1:65]) )
     shared = data[1:1 + PUBKEY_LEN]
-    key_material = ecdh_agree(privkey, keys.PublicKey(shared))
+    try:
+        key_material = ecdh_agree(privkey, keys.PublicKey(shared))
+    except ValueError as exc:
+        raise DecryptionError(
```
Comment (pipermerriam) [inline p2p/ecies.py]: Requesting comment explaining why this happens.
